### ISSUE TITLE
[doc] Improve documentation on usage of violationSuppressXPath

### ISSUE THREAD

**Issue by mfvanek** [pmd/pmd#4901]
**Is your feature request related to a problem? Please describe.**
While migrating to PMD 7 I faced with extreme troubles how to rewrite old suppressions to a new style.
Example from docs doesn't work for me https://docs.pmd-code.org/pmd-doc-7.0.0/pmd_userdocs_suppressing_warnings.html
```
<!-- doesn't work in PMD 7 -->
<property name="violationSuppressXPath" value="./ancestor::ClassDeclaration[contains(@Image, 'Bean')]"/>
```

**Describe the solution you'd like**
A good documentation with working examples

I found a clue in https://github.com/pmd/pmd/issues/4707

**Comment by mfvanek** [comment]
Example 1
```
    <rule ref="category/java/bestpractices.xml/AbstractClassWithoutAbstractMethod">
        <properties>
            <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration['.*TestBase']"/>
        </properties>
    </rule>
```
transformed to

```
    <rule ref="category/java/bestpractices.xml/AbstractClassWithoutAbstractMethod">
        <properties>
            <property name="violationSuppressXPath" value=".[@SimpleName = 'TestBase'] | .[@SimpleName = 'BaseEntity']"/>
        </properties>
    </rule>
```

Example 2
```
    <rule ref="category/java/codestyle.xml/LinguisticNaming">
        <properties>
            <property name="checkSetters" value="false"/>
            <property name="violationSuppressXPath"
                      value="//ClassOrInterfaceBodyDeclaration/Annotation/MarkerAnnotation//Name[@Image='Test']"/>
        </properties>
    </rule>
```
transformed to 
```
    <rule ref="category/java/codestyle.xml/LinguisticNaming">
        <properties>
            <property name="checkSetters" value="false"/>
            <property name="violationSuppressXPath"
                      value=".[pmd-java:hasAnnotation('org.junit.jupiter.api.Test')]"/>
        </properties>
    </rule>
```

**Comment by zman0900** [comment]
Example 1 above doesn't work for me either.  To exclude errors in classes with names ending in "Test", I found that this works:
```xml
<property name="violationSuppressXPath" value="./ancestor::ClassDeclaration[matches(@SimpleName, '^.*Test$')]" />
```
